ClickHouse.Client/Types/JsonType.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using ClickHouse.Client.Formats;
+using ClickHouse.Client.Types.Grammar;
+
+namespace ClickHouse.Client.Types;
+
+internal class JsonType : ParameterizedType
+{
+    public override Type FrameworkType => typeof(string);
+
+    public override string Name => "Json";
+
+    public override object Read(ExtendedBinaryReader reader)
+    {
+        var value = new JsonObject();
+        var nfields = reader.Read7BitEncodedInt();
+
+        for (int i = 0; i < nfields; i++)
+        {
+            var fieldName = reader.ReadString();
+            // See https://github.com/ClickHouse/ClickHouse/blob/b618fe03bf96e64bea1a1bdec01adc1c00cd61fb/src/DataTypes/DataTypesBinaryEncoding.cpp#L48
+            // https://clickhouse.com/docs/en/sql-reference/data-types/data-types-binary-encoding
+            // TODO: add type codes within types themselves
+            var typeCode = reader.Read7BitEncodedInt();
+            object fieldValue = typeCode switch
```

Comment:
Consider using `JsonValue.Create` in each switch statement and avoid the boxing here.